library/x509_create.c

```diff
@@ -123,79 +130,206 @@ static const x509_attr_descriptor_t *x509_attr_descr_from_name(const char *name,
     return cur;
 }
 
+static int hex_to_int(char c)
+{
+    return ('0' <= c && c <= '9') ? (c - '0') :
+           ('a' <= c && c <= 'f') ? (c - 'a' + 10) :
+           ('A' <= c && c <= 'F') ? (c - 'A' + 10) : -1;
+}
+
+static int hexpair_to_int(char c1, char c2)
```

Comment:
`hexpair_to_int()` is only ever called with `*p` and `*(p + 1)`, so would it be better to pass just `p` to it?